lib/index.js

```diff
@@ -122,6 +128,37 @@ internals.handler = function (route, options) {
             }
         }
 
-        return responder.view(settings.template, context, settings.options);
+        return h.view(settings.template, context, settings.options);
     };
 };
+
+
+internals.state = (realm) => {
+
+    const state = realm.plugins.vision = realm.plugins.vision || {};
+    return state;
+};
+
+
+internals.nearestState = function (realm) {
+
+    if (realm.plugins.vision) {
+        return realm.plugins.vision;
+    }
+
+    const parent = realm.parent;
+
+    return internals.nearestState(parent);
```

Comment:
Perhaps (?) not a practical concern here, but beware recursion in javascript.  There's still no such thing as tail recursion in node, so each time you recurse the call stack gets longer (and it has a max allowed size).

Reply:
(And this can always be expressed as a while loop!)

Reply:
I was having trouble getting code coverage with the while loop Eran had here, because `parent` would never evaluate to false, since the root server _always_ has vision state. The root server always has vision state because of the `{ setup: true }` introduced in the `register` func.

Here's what's currently in vision - the loop Eran wrote:
```
internals.realm = function (server) {

    if (server.realm.plugins.vision) {
        return server.realm.plugins.vision;
    }

    let parent = server.realm.parent;
    while (parent) {
        if (parent.plugins.vision) {
            return parent.plugins.vision;
        }

        parent = parent.parent;
    }

    return {};
};
```

As soon as the root server goes in this loop, it is returned as having a realm because setup.true will always be there. So `parent = parent.parent` will never run, which would cause it to evaluate to false in the while loop, and return `{}`

Reply:
This could possibly be rewritten as `nearestManager`, which would fix the issue of returning just because a realm has state in vision